hoomd/md/mesh/bond.py

```diff
@@ -0,0 +1,251 @@
+    """
+
+    def __init__(self, mesh):
+        self._mesh = validate_mesh(mesh)
+
+    def _add(self, simulation):
+        # if mesh was associated with multiple pair forces and is still
+        # attached, we need to deepcopy existing mesh.
+        mesh = self._mesh
+        if (not self._attached and mesh._attached
+                and mesh._simulation != simulation):
+            warnings.warn(
+                f"{self} object is creating a new equivalent mesh structure."
+                f" This is happending since the force is moving to a new "
+                f"simulation. To supress the warning explicitly set new mesh.",
+                RuntimeWarning)
+            self._mesh = copy.deepcopy(mesh)
+        # We need to check if the force is added since if it is not then this is
+        # being called by a SyncedList object and a disagreement between the
+        # simulation and mesh._simulation is an error. If the force is added
+        # then the mesh is compatible. We cannot just check the mesh's _added
+        # property because _add is also called when the SyncedList is synced.
+        elif (not self._added and mesh._added
+              and mesh._simulation != simulation):
+            raise RuntimeError(
+                f"Mesh associated with {self} is associated with "
+                f"another simulation.")
+        super()._add(simulation)
+        # this ideopotent given the above check.
+        self._mesh._add(simulation)
+        # This is ideopotent, but we need to ensure that if we change
+        # neighbor list when not attached we handle correctly.
+        self._add_dependency(self._mesh)
+
+    def _attach(self):
+        """Create the c++ mirror class."""
+        if not self._mesh._added:
+            self._mesh._add(self._simulation)
+        else:
+            if self._simulation != self._mesh._simulation:
+                raise RuntimeError("{} object's mesh is used in a "
+                                   "different simulation.".format(type(self)))
+
+        if not self.mesh._attached:
+            self.mesh._attach()
+
+        if isinstance(self._simulation.device, hoomd.device.CPU):
+            cpp_cls = getattr(_md, self._cpp_class_name)
+        else:
+            cpp_cls = getattr(_md, self._cpp_class_name + "GPU")
+
+        self._cpp_obj = cpp_cls(self._simulation.state._cpp_sys_def,
+                                self._mesh._cpp_obj)
+
+        self._apply_param_dict()
+        self._apply_typeparam_dict_with_mesh(self._cpp_obj, self.mesh)
+
+    def _apply_typeparam_dict_with_mesh(self, cpp_obj, mesh):
+        for typeparam in self._typeparam_dict.values():
+            try:
+                typeparam._attach_with_mesh(cpp_obj, mesh)
+            except ValueError as err:
+                raise err.__class__(
+                    f"For {type(self)} in TypeParameter {typeparam.name} "
+                    f"{str(err)}")
```

Comment:
If we are going to overwrite this, than we should do so with the same name to indicate what we are doing. Then we can call `super()._attach()`